yax/refined/src/main/scala/doobie/refined/package.scala

```diff
@@ -0,0 +1,43 @@
+package doobie
+
+package object refined {
+
+  import eu.timepit.refined.api.{Refined, Validate}
+  import eu.timepit.refined.auto._
+  import eu.timepit.refined._
+  import scala.reflect.runtime.universe.TypeTag
+
+  import doobie.imports._
+  import doobie.util.invariant._
+
+  implicit def refinedMeta[T, P](implicit metaT: Meta[T], v: Validate[T, P], tag: TypeTag[T Refined P]): Meta[T Refined P] =
```

Comment:
Thanks @fthomas! 
My current problem is making it null-safe. My (maybe naive) attempt was 
`  implicit def refinedMeta[T >: Null, P, F[T,P] >: Null](implicit metaT: Meta[T], v: Validate[T, P], refType: RefType[F], ev: Null <:< T, tag: Manifest[F[T,P]]): Meta[F[T,P]] =
    metaT.xmap[F[T,P]](
      t => if (t == null) null else rightOrException[F[T,P]](refType.refine[P](t)(v)),
      ftp => if (ftp == null) ev(null) else refType.unwrap(ftp)
    )
`
but with this the compiler is not able to find a typeclass for types like `type PositiveInt = Int Refined Positive`